test/functional-api/integration/callouts/close-state-callouts.it-spec.ts

```diff
@@ -0,0 +1,655 @@
+import { deleteOrganization } from '../organization/organization.request.params';
+import { removeHub } from '../hub/hub.request.params';
+import { entitiesId } from '@test/functional-api/zcommunications/communications-helper';
+import { uniqueId } from '@test/utils/mutations/create-mutation';
+import {
+  createChallengeWithUsers,
+  createOpportunityWithUsers,
+  createOrgAndHubWithUsers,
+} from '@test/functional-api/zcommunications/create-entities-with-users-helper';
+import {
+  createCalloutOnCollaboration,
+  deleteCallout,
+  getChallengeCalloutByNameId,
+  getHubCalloutByNameId,
+  getOpportunityCalloutByNameId,
+  updateCallout,
+  updateCalloutVisibility,
+} from './callouts.request.params';
+import {
+  AspectTypes,
+  createAspectOnCallout,
+  getDataPerChallengeCallout,
+  getDataPerHubCallout,
+  getDataPerOpportunityCallout,
+} from '../aspect/aspect.request.params';
+import { CalloutState, CalloutType, CalloutVisibility } from './callouts-enum';
+import { delay, TestUser } from '@test/utils';
+import { postCommentInCallout } from '../comments/comments.request.params';
+import { mutation } from '@test/utils/graphql.request';
+import {
+  removeComment,
+  removeCommentVariablesData,
+  sendComment,
+  sendCommentVariablesData,
+} from '@test/utils/mutations/communications-mutation';
+import { createCanvasOnCallout } from '../canvas/canvas.request.params';
+
+let opportunityName = 'aspect-opp';
+let challengeName = 'aspect-chal';
+let calloutNameID = '';
+let calloutDisplayName = '';
+let calloutId = '';
+let cardNameID = '';
+let msessageId = '';
+
+const organizationName = 'callout-org-name' + uniqueId;
+const hostNameId = 'callout-org-nameid' + uniqueId;
+const hubName = 'callout-eco-name' + uniqueId;
+const hubNameId = 'callout-eco-nameid' + uniqueId;
+
+beforeAll(async () => {
+  await createOrgAndHubWithUsers(
+    organizationName,
+    hostNameId,
+    hubName,
+    hubNameId
+  );
+  await createChallengeWithUsers(challengeName);
+  await createOpportunityWithUsers(opportunityName);
+});
+
+afterAll(async () => {
+  await removeOpportunity(entitiesId.opportunityId);
+  await removeChallenge(entitiesId.challengeId);
+  await removeHub(entitiesId.hubId);
+  await deleteOrganization(entitiesId.organizationId);
+});
+
+beforeEach(async () => {
+  challengeName = `testChallenge ${uniqueId}`;
+  opportunityName = `opportunityName ${uniqueId}`;
+  calloutNameID = `callout-name-id-${uniqueId}`;
+  calloutDisplayName = `callout-d-name-${uniqueId}`;
+  cardNameID = `aspect-name-id-${uniqueId}`;
+});
+
+describe('Callouts - Close State', () => {
+  afterEach(async () => {
+    await deleteCallout(calloutId);
+  });
+  test('Close callout that has not been published', async () => {
+    // Act
+    const res = await createCalloutOnCollaboration(
+      entitiesId.hubCollaborationId,
+      calloutDisplayName,
+      calloutNameID
+    );
+    calloutId = res.body.data.createCalloutOnCollaboration.id;
+
+    await updateCallout(
```

Comment:
most of the parameters in this mutation are optional and you are forcing yourself to send them nevertheless.